Port to project.json/xproj and add .NET Standard support

jeremymeng opened the pull request.

jeremymeng committed:
Port to project.json/xproj and add .NET Standard support

jeremymeng commented on src/Serilog.Sinks.NLog/project.json:
```diff
@@ -0,0 +1,27 @@
+﻿{
+  "version": "2.0.0-*",
+  "description": "Serilog event sink that writes to NLog. Merge your new Serilog event stream into your existing NLog infrastructure.",
```
I left the "-*" part here since NLog v4.4.0 is still in beta.

nblumhardt commented on Build.ps1:
```diff
@@ -1,85 +1,15 @@
-    [Switch] $notouch,
-    [String] $sln                  # e.g serilog-sink-name
-)
+Push-Location $PSScriptRoot
```
This script is slightly older than the one in serilog/serilog (there were some issues with build number generation). Can you please update with the latest Build.ps1 from the main Serilog repo?

jeremymeng replied:
Thank you @nblumhardt for reviewing this.

I see there are [test sections](https://github.com/serilog/serilog/blob/dev/Build.ps1#L29-L49) which don't apply to this repro.  I could either add a check for the existence of `test` folder, or remote the sections for this repo.  Which approach do you prefer?

nblumhardt replied:
The existence check would be fine; where possible I've tried adding empty test projects in these cases, which is probably the ideal option if it's not too much trouble.

nblumhardt commented on src/Serilog.Sinks.NLog/project.json:
```diff
@@ -0,0 +1,27 @@
+﻿{
+  "version": "2.0.0-*",
```
There's already a 2.x of this sink published - can we update this to 3.0.0-\* please?

nblumhardt commented:
Thanks Jeremy, this is looking great. Can you please re-target the PR to the `dev` branch? I just noticed this repo is missing one, so I've created it now :-)

jeremymeng committed:
Address CR feedback

- Set version to v3 as there's already v2 releases.
- Update Build.ps1
- Add empty test directory

jeremymeng committed:
Build using preview2 tooling

jeremymeng commented:
Closing this, will open a new one targeting `dev` branch.

nblumhardt commented:
Thanks. Just a heads-up, the `Edit` button next to the title in a PR now lets you change the target branch without closing/reopening.

jeremymeng commented:
Ah didn't know that.  Thanks a lot for the info!
